src/main/java/uk/gov/register/resources/v2/BlobResource.java

```diff
@@ -58,11 +69,21 @@ public ItemView getBlobDataByHex(@PathParam("blob-hash") String blobHash) throws
             ExtraMediaType.TEXT_CSV,
     })
     @Timed
-    public BlobListView listBlobs() throws FieldConversionException {
-        Collection<Item> items = register.getAllItems(EntryType.user);
+    public BlobListView listBlobs(@QueryParam("start") Optional<IntegerParam> optionalStart, @QueryParam("limit") Optional<IntegerParam> optionalLimit) throws FieldConversionException {
+        int start = optionalStart.map(IntParam::get).orElse(1);
+        int limit =  optionalLimit.map(IntParam::get).orElse(100);
 
-        // TODO: allow this resource to be paginated
-        return buildBlobListView(items.stream().limit(100).collect(Collectors.toList()));
+        Collection<Item> items = register.getUserItemsPaginated(start, limit + 1);
+
+        boolean hasNext = (items.size() == limit + 1);
+
+        if(hasNext) {
+            Item finalBlob = items.stream().skip(limit).findFirst().get();
```

Comment:
it is dangerous to call `.get()` on an `Optional` can we do something similar to https://github.com/openregister/openregister-java/blob/292138e102e760434a987fcc1cef851ba98110e8/src/main/java/uk/gov/register/resources/v1/ItemResource.java#L45 ?

Reply:
We could but this should never get called if you're falling into this block because it checks the size first. I'm just trying to access the last item in the list here but I couldn't figure out a nice way to do it. Is there a cleaner way to rewrite this whole block?